Notation.Plugin.AzureKeyVault/KeyVault/KeyVaultClient.cs

```diff
@@ -148,10 +151,6 @@ private static KeyVaultMetadata ParseId(string id)
         public async Task<byte[]> SignAsync(SignatureAlgorithm algorithm, byte[] payload)
         {
             var signResult = await _cryptoClient.Value.SignDataAsync(algorithm, payload);
-            if (signResult.KeyId != _keyId)
-            {
-                throw new PluginException($"Invalid keys identifier. The user provides {_keyId} but the response contains {signResult.KeyId} as the keys");
-            }
```

Comment:
What's the reason of removing this check?

Reply:
There are 2 cases:
for versoned id, I think we need it.
for versionless id, I think we dont't need it.

OK, let me restore it for versioned id.